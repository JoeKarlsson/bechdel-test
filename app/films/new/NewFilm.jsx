import React from 'react';
import { Link } from 'react-router';
import styles from './NewFilm.scss';
import * as $ from'jquery';
import Loading from '../../shared/loading/Loading.jsx'

class NewFilm extends React.Component {
  constructor() {
    super();
    this.state = {
      film: '',
      fileName: '',
      blob: '',
      isLoading: false,
    };
    this.handleScriptUploadChange = this.handleScriptUploadChange.bind(this)
    this.handleFilmSubmit = this.handleFilmSubmit.bind(this)
  }

  handleFilmSubmit(e) {
    e.preventDefault();
    this.setState({ isLoading: true });
    let fd = new FormData(document.querySelector('form'));
    $.ajax({
      url: '/api/film',
      method: 'POST',
      data: fd,
      cache: false,
      processData: false,
      contentType: false,
      success: function(data) {
        console.log(data, 'SUCCESS');
        window.location = '/film/' + data._id;
      }.bind(this),
      error: function(xhr, status, err) {
        this.setState({ loading: false });
        console.error(status, err.toString());
      }.bind(this)
    });
  }

  handleScriptUploadChange(e){
    var reader = new FileReader();
    var file = e.target.files[0];

    reader.onload = (upload) => {
      this.setState({
        blob: upload.target.result,
        fileName: file.name,
      });
    }

    if(file){
      reader.readAsDataURL(file);

    }
  }

  render() {
    return (
      <div className={styles.newFilm}>
        <h1>test a new script</h1>
        {
          this.state.isLoading ?
          <Loading />
        :
          <form
            name='script'
            action='/api/film'
            encType='multipart/form-data'
            method='POST'
          >
            <p>
              Note: This tool currently only suppports scripts with a .txt format, and the script must follow the <a target="blank" href='http://www.simplyscripts.com/WR_format.html'>Standard Script Format</a>.
            </p>
            <p>
              If you are still having issues using the tool, try adding the title of the script is on the first line. Or you can submit an <a target="blank" href="https://github.com/JoeKarlsson1/bechdel-test/issues">issue</a>.
            </p>
            <p>
              You can check out a example script <a target="blank" href='https://github.com/JoeKarlsson1/bechdel-test/blob/master/tests/server/methods/test-script.txt'>here</a>.
            </p>
            <p>
              Please specify a file, or a set of files:<br />
            </p>
            <input
              type='file'
              name='script'
              size='40'
              onChange={this.handleScriptUploadChange}
            />
            <div>
              <button onClick={this.handleFilmSubmit}>Submit Script</button>
            </div>
          </form>
        }
      </div>
    )
  }
}

NewFilm.propTypes = {
  film: React.PropTypes.string,
  fileName: React.PropTypes.string,
  blob: React.PropTypes.string,
};

NewFilm.defaultProps = {
  film: '',
  fileName: '',
  blob: '',
  isLoading: false,
};

export default NewFilm;
