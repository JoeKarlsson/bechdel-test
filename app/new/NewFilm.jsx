/* jshint esversion: 6 */
// https://developer.mozilla.org/en-US/docs/Web/API/FormData/Using_FormData_Objects
// https://stackoverflow.com/questions/4856917/jquery-upload-progress-and-ajax-file-upload#494377
'use strict';

import React from 'react';
import styles from './NewFilm.scss';
import * as $ from'jquery';

export default React.createClass({
  getInitialState() {
    return {
      film: '',
      fileName: '',
      blob: ''
    }
  },
  handleFilmSubmit(e) {
    e.preventDefault();
    let fd = new FormData(document.querySelector("form"));
    $.ajax({
      url: "/api/film",
      method: 'POST',
      data: fd,
      cache: false,
      processData: false,
      contentType: false,
      success: function(data) {
        console.log(data, 'SUCCESS');
        window.location = '/film/' + data[0]._id;
      }.bind(this),
      error: function(xhr, status, err) {
        console.error(status, err.toString());
      }.bind(this)
    });
  },
  handleScriptUploadChange(e){
    var reader = new FileReader();
    var file = e.target.files[0];

    reader.onload = (upload) => {
      this.setState({blob: upload.target.result, fileName: file.name});
    }

    if(file){
      reader.readAsDataURL(file);

    }
  },
  render() {
    return (
      <div>
        <div>New Film</div>
        <form
          name="script"
          action="/api/film"
          encType="multipart/form-data"
          method="POST"
        >
          <p>Please specify a file, or a set of files:<br /></p>
          <input
            type="file"
            name="script"
            size="40"
            onChange={this.handleScriptUploadChange}
          />
          <div>
            <button onClick={this.handleFilmSubmit}>Submit Script</button>
          </div>
        </form>
      </div>
    )
  }
});