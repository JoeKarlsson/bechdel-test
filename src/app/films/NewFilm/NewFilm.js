import React from 'react';
import postFilmScript from '../../helper/postFilmScript';
import Loading from '../../shared/Loading/Loading';

class NewFilm extends React.Component {
	constructor() {
		super();
		this.state = {
			film: '',
			fileName: '',
			blob: '',
			isLoading: false,
		};
		this.handleScriptUploadChange = this.handleScriptUploadChange.bind(this);
		this.handleFilmSubmit = this.handleFilmSubmit.bind(this);
	}

	handleFilmSubmit(e) {
		e.preventDefault();
		this.setState({ isLoading: true });
		const fd = new FormData(document.querySelector('form'));
		postFilmScript(fd);
	}

	handleScriptUploadChange(e) {
		const reader = new FileReader();
		const file = e.target.files[0];

		reader.onload = (upload) => {
			this.setState({
				blob: upload.target.result,
				fileName: file.name,
			});
		};

		if (file) {
			reader.readAsDataURL(file);
		}
	}

	render() {
		return (
			<div className="newFilm">
				<h1>test a new script</h1>
				{this.state.isLoading ? (
					<Loading />
				) : (
					<form
						name="script"
						action="/api/film"
						encType="multipart/form-data"
						method="POST"
					>
						<p>
						Note: This tool currently only suppports scripts with a .txt
						format, and the script must follow the{' '}
							<a
								target="blank"
								href="http://www.simplyscripts.com/WR_format.html"
							>
							Standard Script Format
							</a>.
						</p>
						<p>
						If you are still having issues using the tool, try adding the
						title of the script is on the first line. Or you can submit an{' '}
							<a
								target="blank"
								href="https://github.com/JoeKarlsson1/bechdel-test/issues"
							>
							issue
							</a>.
						</p>
						<p>
						You can check out a example script{' '}
							<a
								target="blank"
								href="https://github.com/JoeKarlsson1/bechdel-test/blob/master/tests/server/methods/test-script.txt"
							>
							here
							</a>.
						</p>
						<p>
						Please specify a file, or a set of files:<br />
						</p>
						<input
							type="file"
							name="script"
							size="40"
							onChange={this.handleScriptUploadChange}
						/>
						<div>
							<button onClick={this.handleFilmSubmit}>
								Submit Script
							</button>
						</div>
					</form>
				)}
			</div>
		);
	}
}

NewFilm.propTypes = {};

NewFilm.defaultProps = {};

export default NewFilm;
