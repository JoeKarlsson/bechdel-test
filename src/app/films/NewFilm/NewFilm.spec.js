import React from 'react';
import { shallow, configure } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';
import renderer from 'react-test-renderer';
import NewFilm from './NewFilm';

jest.mock('./Uploader/Uploader');

// jest.mock('uppy');
// jest.mock('uppy/lib');
// jest.mock('uppy/lib/core');
// jest.mock('uppy/lib/core/Core.js');
// jest.mock('uppy/lib/plugins/XHRUpload');
// jest.mock('uppy/lib/react');
// jest.mock('uppy/lib/react/DragDrop');

global.window = {
	navigator: {
		onLine: true,
	},
};

configure({ adapter: new Adapter() });

describe('NewFilm', () => {
	let wrapper;

	beforeEach(() => {
		wrapper = shallow(<NewFilm />);
	});

	describe('rendering', () => {
		describe('initial state', () => {
			it('is rendered', () => {
				const component = renderer.create(<NewFilm />);
				const tree = component.toJSON();
				expect(tree).toMatchSnapshot();
			});
			it('is rendered correctly', () => {
				expect(wrapper).toHaveLength(1);
			});

			it('should have correct inital instance', () => {
				const initialInstance = wrapper.instance();
				const expectedInstance = null;
				expect(initialInstance).toBe(expectedInstance);
			});
		});
	});
});
