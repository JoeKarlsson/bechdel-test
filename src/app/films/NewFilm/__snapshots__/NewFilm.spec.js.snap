// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`NewFilm rendering initial state is rendered 1`] = `
<div
  className="newFilm container"
>
  <h1>
    test a new script
  </h1>
  <div>
    Mock Uploader
  </div>
  <p>
    Note: This tool currently only suppports scripts with a .txt format, and the script must follow the
     
    <a
      href="http://www.simplyscripts.com/WR_format.html"
      target="blank"
    >
      Standard Script Format
    </a>
    .
  </p>
  <p>
    If you are still having issues using the tool, try adding the title of the script is on the first line. Or you can submit an
     
    <a
      href="https://github.com/JoeKarlsson1/bechdel-test/issues"
      target="blank"
    >
      issue
    </a>
    .
  </p>
</div>
`;
